src/proposals/20250529/SparkEthereum_20250529.sol

```diff
@@ -0,0 +1,239 @@
+
+import { AllocatorBuffer } from 'dss-allocator/src/AllocatorBuffer.sol';
+import { AllocatorVault }  from 'dss-allocator/src/AllocatorVault.sol';
+
+import { IERC20 }   from 'forge-std/interfaces/IERC20.sol';
+import { IERC4626 } from 'forge-std/interfaces/IERC4626.sol';
+
+import { Ethereum } from 'spark-address-registry/Ethereum.sol';
+import { Optimism } from 'spark-address-registry/Optimism.sol';
+import { Unichain } from 'spark-address-registry/Unichain.sol';
+
+import { MainnetController }               from "spark-alm-controller/src/MainnetController.sol";
+import { RateLimitHelpers, RateLimitData } from "spark-alm-controller/src/RateLimitHelpers.sol";
+
+import { ISparkLendFreezerMom } from 'sparklend-freezer/interfaces/ISparkLendFreezerMom.sol';
+
+import { SparkPayloadEthereum, IEngine, EngineFlags } from "../../SparkPayloadEthereum.sol";
+
+import { IMetaMorpho, MarketParams } from 'metamorpho/interfaces/IMetaMorpho.sol';
+
+import { CCTPForwarder } from 'xchain-helpers/forwarders/CCTPForwarder.sol';
+
+interface IOptimismTokenBridge {
+    function bridgeERC20To(
+        address _localToken,
+        address _remoteToken,
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    ) external;
+}
+
+/**
+ * @title  May 29, 2025 Spark Ethereum Proposal
+ * @notice Spark Liquidity Layer:
+ *         - Onboard Optimism, Unichain
+ *         - Mint 100m USDS into Optimism and Unichain
+ *         - Mint 100m sUSDS into Optimism and Unichain
+ *         - Increase USDe Mint and Swap Rate Limits
+ *         SparkLend:
+ *         - Update USDS IRM
+ *         - Update DAI IRM
+ *         - Reduce WBTC Liquidation Threshold
+ *         Spark DAI Vault:
+ *         - Onboard PT-SUSDS-14Aug2025
+ * @author Phoenix Labs
+ * Forum:  https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell/26372/2
+ *         https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell-2/26440
+ * Vote:   TODO
+ */
+contract SparkEthereum_20250529 is SparkPayloadEthereum {
+
+    address internal constant DAI_USDS_IRM                  = 0xE15718d48E2C56b65aAB61f1607A5c096e9204f1;  // DAI  and USDS use the same params, same IRM
+    
+    address internal constant PT_SUSDS_14AUG2025            = 0xFfEc096c087C13Cc268497B89A613cACE4DF9A48;
+    address internal constant PT_SUSDS_14AUG2025_PRICE_FEED = 0xD7c8498fF648CBB9E79d6470cf7F639e696D27A5;
+
+    uint256 internal constant SUSDS_BRIDGE_AMOUNT  = 100_000_000e18;
+    uint256 internal constant SUSDS_DEPOSIT_AMOUNT = 200_000_000e18;
+    uint256 internal constant USDS_BRIDGE_AMOUNT   = 100_000_000e18;
+    uint256 internal constant USDS_MINT_AMOUNT     = 400_000_000e18;
+
+    constructor() {
+        PAYLOAD_BASE     = 0x08AbA599Bd82e4De7b78516077cDF1CB24788CC1;
+        PAYLOAD_OPTIMISM = 0x08AbA599Bd82e4De7b78516077cDF1CB24788CC1;
+        PAYLOAD_UNICHAIN = 0xbF5a7CfaF47fd1Ad75c9C613b1d4C196eE1b4EeF;
```

Comment:
These need to be redeployed with fixes I mentioned.

Reply:
redeployed